### ISSUE TITLE
Flaky extension test

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#20255]
I see this one failing very often, but not reliably (e.g. https://ci.appveyor.com/project/VSCode/vscode/build/1.0.14059#L3039):

```
Invalid Version: undefined
Error: oops
    at ErrorHandler.unexpectedErrorHandler (C:\projects\vscode\test\all.js:258:12)
    at ErrorHandler.onUnexpectedError (C:\projects\vscode\out\vs\base\common\errors.js:42:18)
    at Object.onUnexpectedError (C:\projects\vscode\out\vs\base\common\errors.js:60:34)
    at CallbackList.invoke (C:\projects\vscode\out\vs\base\common\callbackList.js:60:30)
    at Emitter.fire (C:\projects\vscode\out\vs\base\common\event.js:88:40)
    at ExtensionsWorkbenchService.syncLocalWithGalleryExtension (C:\projects\vscode\out\vs\workbench\parts\extensions\node\extensionsWorkbenchService.js:452:28)
    at ExtensionsWorkbenchService.fromGallery (C:\projects\vscode\out\vs\workbench\parts\extensions\node\extensionsWorkbenchService.js:444:26)
    at C:\projects\vscode\out\vs\workbench\parts\extensions\node\extensionsWorkbenchService.js:404:110
    at Array.map (native)
    at Object.mapPager (C:\projects\vscode\out\vs\base\common\paging.js:88:40)
    at C:\projects\vscode\out\vs\workbench\parts\extensions\node\extensionsWorkbenchService.js:404:59
    at CompletePromise_ctor.CompletePromise_then [as then] (C:\projects\vscode\out\vs\base\common\winjs.base.raw.js:1566:49)
    at ExtensionsWorkbenchService.queryGallery (C:\projects\vscode\out\vs\workbench\parts\extensions\node\extensionsWorkbenchService.js:404:18)
    at ExtensionsWorkbenchService.syncWithGallery (C:\projects\vscode\out\vs\workbench\parts\extensions\node\extensionsWorkbenchService.js:475:25)
    at loop (C:\projects\vscode\out\vs\workbench\parts\extensions\node\extensionsWorkbenchService.js:462:51)
    at Throttler.queue (C:\projects\vscode\out\vs\base\common\async.js:113:34)
    at C:\projects\vscode\out\vs\base\common\async.js:230:93
    at C:\projects\vscode\out\vs\base\common\async.js:186:28
    at Object.notifySuccess [as _notify] (C:\projects\vscode\out\vs\base\common\winjs.base.raw.js:1170:59)
    at Object.enter (C:\projects\vscode\out\vs\base\common\winjs.base.raw.js:852:30)
    at Promise_ctor._run (C:\projects\vscode\out\vs\base\common\winjs.base.raw.js:1068:29)
    at Promise_ctor._completed (C:\projects\vscode\out\vs\base\common\winjs.base.raw.js:1036:18)
    at Timeout._onTimeout (C:\projects\vscode\out\vs\base\common\async.js:191:23)
    at tryOnTimeout (timers.js:232:11)
    at Timer.listOnTimeout (timers.js:202:5)

```

**Comment by sandy081** [comment]
Well.. as described very flaky.. Can't reproduce it consistently but once in 10 times. So providing a fix by analysing the logs. Suspecting that the delayer task that syncs gallery with local is still running even after extensions workbench service is disposed. Added a step to stop the delayer when service is being disposed. This is semantically correct. 

I did a trial run of 15 times with fix and I do not see this error. Assuming this is the root cause.
